apis/hive/v1/hiveconfig_types.go

```diff
@@ -250,6 +260,25 @@ type AWSPrivateLinkSubnet struct {
 	AvailabilityZone string `json:"availabilityZone"`
 }
 
+// GCPPrivateServiceConnectConfig defines the gcp private service connect config for th eprivate-link controller.
+type GCPPrivateServiceConnectConfig struct {
+	// CredentialsSecretRef references a secret in the TargetNamespace that will be used to authenticate with
+	// GCP for creating the resources for GCP Private Service Connect
+	CredentialsSecretRef corev1.LocalObjectReference `json:"credentialsSecretRef"`
+
+	// EndpointSubnetInventory is a list of subnets and their corrisponding regions.
+	// The controller uses this list to choose a subnet for creating Private Service Connect Endpoints.
+	// Since the Endpoints must be in the same region as the ClusterDeployment, we must have subnets in that
+	// region to be able to setup Private Service Connect.
+	EndpointSubnetInventory []GCPPrivateServiceConnectInventory `json:"endpointSubnetInventory,omitempty"`
```

Comment:
Should we inject a VPC level here that enables per-vpc credentials (inline with how AWS does it?)

Reply:
That may be a question for the SD folks consuming this. Do they use the granular cloud secrets for PrivateLink?

I may need some help understanding the connection, if any, between the VPCs in EndpointVPCInventory and AssociatedVPCs, and what the equivalent is for GCP.

Reply:
I see... I was mixing which one had the credential. The AssociatedVPCs has the credential while the EndpointVPCInventory does not. I am going to inject a VPC (network) into the structure to be closer aligned to the AWS counterpart. This will allow us to add credentials per VPC later if needed.

Reply:
I have modified this to be more on par with aws by using EndpointVPCInventory which contains a list of networks and their subnetworks. This enables us to later add credentials to each network if a use case arises.
https://github.com/openshift/hive/compare/f0759b0ad10b8f12ead2fdeead63bbadda7ff245..16122cc3b18a61499e3a4e057202de0c776e6a7a#diff-81dcbf3f7158f9d5074e3f288096c3a4f7f1052a34efa4260f259896ba3374faR297-R313